build: unable to fetch external binaries

<!--  As an open source project with a dedicated but small maintainer team, it can sometimes take a long time for issues to be addressed so please be patient and we will get back to you as soon as we can.
-->

### Preflight Checklist
<!-- Please ensure you've completed the following steps by replacing [ ] with [x]-->

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * <!-- (output of `node_modules/.bin/electron --version`) e.g. 4.0.3 -->6.0.0-nightly.20190311 
* **Operating System:**
  * <!-- (Platform and Version) e.g. macOS 10.13.6 / Windows 10 (1809) / Ubuntu 18.04 x64 --> Windows 10 (1809)
* **Last Known Working Electron version:**:
  * <!-- (if applicable) e.g. 3.1.0 -->n/a

### Expected Behavior
<!-- A clear and concise description of what you expected to happen. -->
`gclient sync` completes successfully

### Actual Behavior
<!-- A clear and concise description of what actually happened. -->
The `update_external_binaries.py` hook fails with this error
```
________ running 'D:\depot_tools\win_tools-2_7_6_bin\python\bin\python.exe src/electron/script/update-external-binaries.py' in 'D:\electron'
Downloading http://github.com/electron/electron-frameworks/releases/download/v1.4.0/directxsdk-x64.zip to c:\users\rictow01\appdata\local\temp\electron-ectopy\directxsdk-x64.zip
Traceback (most recent call last):
  File "src/electron/script/update-external-binaries.py", line 108, in <module>
    main()
  File "src/electron/script/update-external-binaries.py", line 50, in main
    temp_path = download_binary(base_url, version, binary['url'])
  File "src/electron/script/update-external-binaries.py", line 87, in download_binary
    temp_path = download_to_temp_dir(full_url, filename=binary_url)
  File "src/electron/script/update-external-binaries.py", line 94, in download_to_temp_dir
    download(text='Download ' + filename, url=url, path=file_path)
  File "D:\electron\src\electron\script\lib\util.py", line 68, in download
    web_file = urllib2.urlopen(url)
  File "D:\depot_tools\win_tools-2_7_6_bin\python\bin\lib\urllib2.py", line 127, in urlopen
    return _opener.open(url, data, timeout)
  File "D:\depot_tools\win_tools-2_7_6_bin\python\bin\lib\urllib2.py", line 410, in open
    response = meth(req, response)
  File "D:\depot_tools\win_tools-2_7_6_bin\python\bin\lib\urllib2.py", line 523, in http_response
    'http', request, response, code, msg, hdrs)
  File "D:\depot_tools\win_tools-2_7_6_bin\python\bin\lib\urllib2.py", line 442, in error
    result = self._call_chain(*args)
  File "D:\depot_tools\win_tools-2_7_6_bin\python\bin\lib\urllib2.py", line 382, in _call_chain
    result = func(*args)
  File "D:\depot_tools\win_tools-2_7_6_bin\python\bin\lib\urllib2.py", line 629, in http_error_302
    return self.parent.open(new, timeout=req.timeout)
  File "D:\depot_tools\win_tools-2_7_6_bin\python\bin\lib\urllib2.py", line 404, in open
    response = self._open(req, data)
  File "D:\depot_tools\win_tools-2_7_6_bin\python\bin\lib\urllib2.py", line 422, in _open
    '_open', req)
  File "D:\depot_tools\win_tools-2_7_6_bin\python\bin\lib\urllib2.py", line 382, in _call_chain
    result = func(*args)
  File "D:\depot_tools\win_tools-2_7_6_bin\python\bin\lib\urllib2.py", line 1222, in https_open
    return self.do_open(httplib.HTTPSConnection, req)
  File "D:\depot_tools\win_tools-2_7_6_bin\python\bin\lib\urllib2.py", line 1184, in do_open
    raise URLError(err)
urllib2.URLError: <urlopen error [Errno 1] _ssl.c:507: error:1407742E:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert protocol version>
Error: Command 'D:\\depot_tools\\win_tools-2_7_6_bin\\python\\bin\\python.exe src/electron/script/update-external-binaries.py' returned non-zero exit status 1 in D:\electron
```

It's possible that GitHub may be trying to redirect from HTTP to a TLSv1.2-secured HTTPS connection (Python 2.7.6 - which is what ships in depot_tools - is too old to support it).

### To Reproduce
Attempt to build `master` on a Windows 10 system.

--- a/script/external-binaries.json
+++ b/script/external-binaries.json
@@ -1,5 +1,5 @@
 {
-  "baseUrl": "http://github.com/electron/electron-frameworks/releases/download",
+  "baseUrl": "https://github.com/electron/electron-frameworks/releases/download",
   "version": "v1.4.0",
   "binaries": [
     {

